src/components/cc-email-list/cc-email-list.smart.js

```diff
@@ -0,0 +1,242 @@
+              addEmailForm.state = 'idle';
+            });
+          }
+        });
+
+    });
+
+    onEvent('cc-email-list:delete', (address) => {
+
+      updateSecondary(address, (secondaryAddressState) => {
+        secondaryAddressState.state = 'deleting';
+      });
+
+      api.deleteSecondaryEmailAddress(address)
+        .then(() => {
+          notifySuccess(component, i18n('cc-email-list.secondary.action.delete.success', { address }));
+
+          updateComponent('emails', (emails) => {
+            emails.value.secondaryAddresses = emails.value.secondaryAddresses.filter((a) => a.address !== address);
+          });
+        })
+        .catch((error) => {
+          console.error(error);
+          notifyError(component, i18n('cc-email-list.secondary.action.delete.error', { address }));
+          updateSecondary(address, (secondaryAddressState) => {
+            secondaryAddressState.state = 'idle';
+          });
+        });
+    });
+
+    onEvent('cc-email-list:mark-as-primary', (address) => {
+
+      updateSecondary(address, (secondaryAddressState) => {
+        secondaryAddressState.state = 'marking-as-primary';
+      });
+
+      api.markSecondaryEmailAddressAsPrimary(address)
+        .then(() => {
+          notifySuccess(component, i18n('cc-email-list.secondary.action.mark-as-primary.success', { address }));
+
+          const primaryAddress = component.emails.value.primaryAddress.address;
+
+          updateComponent('emails', (emails) => {
+            emails.value.primaryAddress.address = address;
+          });
+          updateSecondary(address, (secondaryAddressState) => {
+            secondaryAddressState.state = 'idle';
+            secondaryAddressState.address = primaryAddress;
+          });
+        })
+        .catch((error) => {
+          console.error(error);
+          notifyError(component, i18n('cc-email-list.secondary.action.mark-as-primary.error', { address }));
+          updateSecondary(address, (secondaryAddressState) => {
+            secondaryAddressState.state = 'idle';
+          });
+        });
+    });
+  },
+});
+
+// -- API calls
+function getApi (apiConfig, signal) {
+  return {
+    fetchEmailAddresses () {
+      return Promise.all([
+        this.fetchPrimaryEmailAddress(),
+        this.fetchSecondaryEmailAddresses(),
+      ]).then(([self, secondary]) => {
+        return { self, secondary };
+      });
+    },
+
+    fetchPrimaryEmailAddress () {
+      return Promise.resolve({
+        method: 'get',
+        url: `/v2/self`,
+        headers: { Accept: 'application/json' },
+      })
+        .then(sendToApi({ apiConfig, signal }));
+    },
+
+    fetchSecondaryEmailAddresses () {
+      return getEmailAddresses()
+        .then(sendToApi({ apiConfig, signal }));
+    },
+
+    sendConfirmationEmail () {
+      return Promise.resolve({
+        method: 'get',
+        url: `/v2/self/confirmation_email`,
+        headers: { Accept: 'application/json' },
+      })
```

Comment:
This one is available on `user` as `todo_getConfirmationEmail`. Can you use it with an alias?

Reply:
thanks